### PR TITLE
add events and vocabulary for the competency-aligned learning profile

### PR DESCRIPTION
**Pull request by joshuakfarrar**

### CODE REVIEW

File: caliper-spec.md [hunk 1]
```diff
@@ -737,6 +742,32 @@ Any Caliper action MAY be used to describe the interaction.
...
+#### Supported Events
+[EvidenceEvent](#evidenceEvent), [MasteryEvent](#masteryEvent)
+
+#### Supported Actors
+[Person](#person)
+
+#### Supported Actions
+
+##### EvidenceEvent 
+[Generated](#generated)
```
Comment (ViktorHaag) [inline caliper-spec.md]: Can we choose a word other than "Generated" here as it creates dissonance with our field names... maybe "Demonstrated" instead? Or something similar to that?

File: caliper-spec.md [hunk 2]
```diff
@@ -1970,7 +2110,95 @@ http://purl.imsglobal.org/caliper/GradeEvent
...
+| membership | [Membership](#membership) &#124; [IRI](#iriDef) | The relationship between the `actor` and the `group` in terms of roles assigned and current status.  The `membership` value MUST be expressed either as an object or as a string corresponding to the membership entity's [IRI](#iriDef). | Optional |
+| session | [Session](#session) &#124; [IRI](#iriDef) | The current user [Session](#session).  The `session` value MUST be expressed either as an object or as a string corresponding to the session's [IRI](#iriDef). | Optional |
+| federatedSession | [LtiSession](#ltiSession) &#124; [IRI](#iriDef) | If the [Event](#event) occurs within the context of an [LTI](#ltiDef) tool launch, the actor's tool consumer [LtiSession](#ltiSession) MAY be referenced.  The `federatedSession` value MUST be expressed either as an object or as a string corresponding to the federatedSession's [IRI](#iriDef). | Optional |
+| extensions | Object | A map of additional attributes not defined by the model MAY be specified for a more concise representation of the [GradeEvent](#gradeEvent). | Optional |
+
+#### Example: GradeEvent (graded)
```
Comment (ViktorHaag) [inline caliper-spec.md]: Should this example not be a Mastery Event?

File: caliper-spec.md [hunk 3]
```diff
@@ -1970,7 +2110,95 @@ http://purl.imsglobal.org/caliper/GradeEvent
...
+[MasteryEvent](#masteryEvent) inherits all properties defined by its supertype [Event](#event). Additional requirements are described below:
+
+| Property | Type | Description | Disposition |
+| :------- | :--- | ----------- | :---------: |
+| id | [UUID](#uuidDef) | The emitting application MUST provision the [Event](#event) with a [UUID](#uuidDef).  A version 4 [UUID](#uuidDef) SHOULD be generated.  The UUID MUST be expressed as a [URN](#urnDef) using the form `urn:uuid:<UUID>` per [RFC 4122](#rfc4122). | Required |
+| type | [Term](#termDef) | The string value MUST be set to the [Term](#termDef) *GradeEvent*. | Required |
+| actor | [Agent](#agent) &#124; [IRI](#iriDef) | An [Agent](#agent), typically [Person](#person) or [SoftwareApplication](#softwareApplication), MUST be specified as the `actor`.  The `actor` value MUST be expressed either as an object or as a string corresponding to the actor's [IRI](#iriDef). | Required |
+| action | [Term](#termDef) | The action or predicate that binds the `actor` or subject to the `object`.  The value range is limited to the action terms listed above.  Only one `action` [Term](#termDef) may be specified per [Event](#event). | Required |
```
Comment (ViktorHaag) [inline caliper-spec.md]: I'm not sure where "above" the action terms allowed for a Mastery Event can be found?

File: caliper-spec.md [hunk 4]
```diff
@@ -3341,7 +3569,41 @@ The following [Chapter](#chapter) properties have been DEPRECATED and MUST NOT b
 }
 ```
 
-### <a name="courseOffering"></a>C.10 CourseOffering
+### <a name="competency"></a>C.10 Competency
+
+A Caliper [Competency](#competency) represents not simply a learning objective, but a skill which the learner should be able to perform if they have mastered it.
```
Comment (ViktorHaag) [inline caliper-spec.md]: How does|should this entity align with IMS CASE?

File: caliper-spec.md [hunk 5]
```diff
@@ -3341,7 +3569,41 @@ The following [Chapter](#chapter) properties have been DEPRECATED and MUST NOT b
...
+| id | [IRI](#iriDef) | A valid [IRI](#iriDef) MUST be specified. The [IRI](#iriDef) MUST be unique and persistent. The [IRI](#iriDef) SHOULD also be dereferenceable, i.e., capable of returning a representation of the resource. A [URI](#uriDef) employing the [URN](#urnDef) scheme MAY be provided in cases where a [Linked Data](#linkedDataDef) friendly HTTP URI is either unavailable or inappropriate. | Required |
+| type | [Term](#termDef) | The string value MUST be set to the [Term](#termDef) *Competency*. | Required |
+| name | string | A name for the competency, likely the skill it represents. | Optional |
+| description | string | A string value comprising a brief, written representation of the [Competency](#competency). | Optional |
+| parent | [Competency](#competency) &#124; [IRI](#iriDef) | If competencies roll up to representations of higher-order skills, the parent property can be used to "point" a competency to that skill. | Optional |
+| level | integer | In the case where competencies roll up to higher-order skills, it is perhaps useful to be able to refer to competencies by their tier, e.g. first-level competencies vs third-level competencies. | Optional |
```
Comment (ViktorHaag) [inline caliper-spec.md]: Is the way of representing this concept portable across competency systems? Would it be better to represent this as a string value, for example?

File: caliper-spec.md [hunk 6]
```diff
@@ -3800,7 +4062,78 @@ The following [EpubVolume](#epubVolume) properties have been DEPRECATED and MUST
...
+[Entity](#entity)
+
+#### Properties
+[Evidence](#evidence) inherits all the properties and requirements defined for its supertype [Entity](#entity).  Additional properties and requirements are described below:
+ 
+| Property | Type | Description | Disposition |
+| :------- | :--- | ----------- | :---------: |
+| id | [IRI](#iriDef) | A valid [IRI](#iriDef) MUST be specified. The [IRI](#iriDef) MUST be unique and persistent. The [IRI](#iriDef) SHOULD also be dereferenceable, i.e., capable of returning a representation of the resource. A [URI](#uriDef) employing the [URN](#urnDef) scheme MAY be provided in cases where a [Linked Data](#linkedDataDef) friendly HTTP URI is either unavailable or inappropriate. | Required |
+| type | [Term](#termDef) | The string value MUST be set to the [Term](#termDef) *Evidence*. | Required |
+| alignedCompetency | [Competency](#competency) &#124; [IRI](#iriDef) | The [Competency](#competency) which the Evidence Event asserts the Actor has completed some measurable criteria towards the demonstration of Mastery. | Optional |
```
Comment (ViktorHaag) [inline caliper-spec.md]: Should this be a list? Can one piece of evidence align to multiple competencies?

**Comment by ViktorHaag** [comment]
How does this proposal align with Caliper's existing ideas about "learning objectives"? Is there a relationship between those and competencies?

File: caliper-spec.md (hunk 1 above)
Comment (ottonomy) [reply caliper-spec.md]: What does Generated really mean in this case? That a learner created and offered to an instructor a work sample? What stage of this process are we talking about?

1. Nate wrote an essay, late into the night, tapping out his deepest close readings of the assigned text on a late model Macbook. 
2. Nate turned in a printed copy of the essay to his instructor by dropping it in the instructor's inbox on her desk.
3. Nate added an electronic version of the essay to his ePortfolio and shared the link as evidence for an assessment of competency mastery.

**Comment by joshuamcghee** [comment]
@ViktorHaag considering the changes made #531  is this pr still relevant/needed?

**Comment by ViktorHaag** [comment]
> @ViktorHaag considering the changes made #531 is this pr still relevant/needed?

Yes it is -- the meat of this PR is to suggest a new profile that tracks learning activity around competencies (i.e. demonstrating master of competencies, and receiving recognition of such). As long as we don't lose sight of the fact that this remains a gap in Caliper and needs addressing, we're OK. Merely providing for a way that CASE can be used piggy-backing on the existing LearningObjective entity isn't sufficient.

**Pull request closed without merging**